### PR TITLE
Expose additional fields in resource_sql_source_representation_instance

### PR DESCRIPTION
**Pull request by rsinnet**
Expose additional fields from the v1beta4 API:
- `username`
- `password`
- `caCertificate`
- `clientCertificate`
- `clientKey`
- `dumpFilePath`

See also: https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1beta4/instances#onpremisesconfiguration

I created a new set of integration tests for the sql_source_representation_instance. However, I could not get tests to pass if I set any of the new fields _besides_ username. I played around with the API and found it does work but does not return some of the values and so it may not be possible to do a terraform `refresh` and obtain these values. I that's what the "ignore" variable was for in the tests, but perhaps I misunderstood. At any rate, I decided to go ahead and post this PR before it goes stale in my brain. **I would be curious if the maintainers happen to know the issue?** I imagine I'm just missing little key piece somewhere. I followed the instructions to build and run the tests and I did manual testing as well with Terraform by using my local build of the provider.

On a side note, it took over 4 hours to understand how to get the integration tests built and running and I had to do some things that weren't documented. Is there somewhere I can/should submit updated documentation to help future developers?

---

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Expose additional API fields in sql_source_representation_instance.
```

### CODE REVIEW

File: mmv1/products/sql/api.yaml [hunk 1]
```diff
@@ -726,6 +726,30 @@ objects:
...
+            description: |
+              PEM representation of the trusted CA's x509 certificate.
+          - !ruby/object:Api::Type::String
+            name: 'client_certificate'
+            description: |
+              PEM representation of the replica's x509 certificate.
+          - !ruby/object:Api::Type::String
+            name: 'client_key'
+            description: |
+              PEM representation of the replica's private key. The corresponsing public key is encoded in the client's certificate.
```
Comment (rileykarson) [inline mmv1/products/sql/api.yaml:748]:
```suggestion
              PEM representation of the replica's private key. The corresponding public key is encoded in the client's certificate.
```

**Review by rileykarson — COMMENTED** [review]
Sorry for the delay in getting to this! Re: integration tests, see https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md for a place where you can suggest changes!

Let me kick off a test run so I can get some logs...

File: mmv1/products/sql/api.yaml (hunk 1 above)
Comment (rsinnet) [reply mmv1/products/sql/api.yaml:748]: 🙈

File: mmv1/products/sql/api.yaml [hunk 2]
```diff
@@ -726,6 +726,30 @@ objects:
...
+          - !ruby/object:Api::Type::String
+            name: 'dump_file_path'
```
Comment (rileykarson) [inline mmv1/products/sql/api.yaml:750]:
```suggestion
            name: 'dumpFilePath'
```

File: mmv1/products/sql/api.yaml [hunk 3]
```diff
@@ -726,6 +726,30 @@ objects:
...
+          - !ruby/object:Api::Type::String
+            name: 'ca_certificate'
+            description: |
+              PEM representation of the trusted CA's x509 certificate.
+          - !ruby/object:Api::Type::String
+            name: 'client_certificate'
+            description: |
+              PEM representation of the replica's x509 certificate.
+          - !ruby/object:Api::Type::String
+            name: 'client_key'
```
Comment (rileykarson) [inline mmv1/products/sql/api.yaml:746]:
```suggestion
            name: 'clientKey'
```

File: mmv1/products/sql/api.yaml [hunk 4]
```diff
@@ -726,6 +726,30 @@ objects:
...
+          - !ruby/object:Api::Type::String
+            name: 'password'
+            description: |
+              The password for connecting to the on-premises instance.
+          - !ruby/object:Api::Type::String
+            name: 'ca_certificate'
+            description: |
+              PEM representation of the trusted CA's x509 certificate.
+          - !ruby/object:Api::Type::String
+            name: 'client_certificate'
```
Comment (rileykarson) [inline mmv1/products/sql/api.yaml:742]:
```suggestion
            name: 'clientCertificate'
```

File: mmv1/products/sql/api.yaml [hunk 5]
```diff
@@ -726,6 +726,30 @@ objects:
...
+          - !ruby/object:Api::Type::String
+            name: 'username'
+            description: |
+              The username for connecting to the on-premises instance.
+          - !ruby/object:Api::Type::String
+            name: 'password'
+            description: |
+              The password for connecting to the on-premises instance.
+          - !ruby/object:Api::Type::String
+            name: 'ca_certificate'
```
Comment (rileykarson) [inline mmv1/products/sql/api.yaml:738]:
```suggestion
            name: 'caCertificate'
```

**Review by rileykarson — COMMENTED** [review]
Weirdly, the API seems to silently accept fields in snake case but return them in camel case. The api.yaml file is camelcased in alignment w/ the API and gets translated into snake case for Terraform automatically.

File: mmv1/products/sql/api.yaml (hunk 2 above)
Comment (rsinnet) [reply mmv1/products/sql/api.yaml:750]: Ah interesting! Sorry I missed this one!

**Comment by rsinnet** [comment]
> I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccPubsubTopicIamBindingGenerated|TestAccPubsubTopicIamMemberGenerated|TestAccPubsubTopicIamPolicyGenerated|TestAccCloudRunService_cloudRunServiceSecretEnvironmentVariablesExample|TestAccCloudRunService_cloudRunServiceSecretVolumesExample|TestAccCloudRunService_secretVolume|TestAccCloudRunService_secretEnvironmentVariable|TestAccCloudFunctionsFunction_vpcConnector|TestAccComputeInstanceFromMachineImage_basic|TestAccComputeInstanceFromMachineImage_overrideMetadataDotStartupScript|TestAccComputeInstanceFromMachineImage_diffProject|TestAccComputeRegionNetworkEndpointGroup_regionNetworkEndpointGroupAppengineExample|TestAccComputeServiceAttachment_serviceAttachmentBasicExampleUpdate|TestAccContainerNodePool_withWorkloadIdentityConfig|TestAccDataprocWorkflowTemplate_basic|TestAccServiceNetworkingPeeredDNSDomain_basic|TestAccOrgPolicyPolicy_FolderPolicy|TestAccOrgPolicyPolicy_OrganizationPolicy|TestAccPrivatecaCertificateTemplate_BasicCertificateTemplate|TestAccPubsubTopic_pubsubTopicBasicExample|TestAccSqlDatabaseInstance_sourceRepresentation_minimal|TestAccSqlDatabaseInstance_sourceRepresentation_full|TestAccSqlDatabaseInstance_withPrivateNetwork|TestAccSQLSourceRepresentationInstance_sqlSourceRepresentationInstanceFromDumpFileExample|TestAccSQLSourceRepresentationInstance_sqlSourceRepresentationInstanceWithSslExample|TestAccSqlUser_postgresIAM|TestAccTPUNode_tpuNodeBUpdateTensorFlowVersion You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=217655

I do not seem to have access to the logs above. I am unable to login with my GitHub credentials.

**Comment by rsinnet** [comment]
> Weirdly, the API seems to silently accept fields in snake case but return them in camel case. The api.yaml file is camelcased in alignment w/ the API and gets translated into snake case for Terraform automatically.

Thanks! Not sure how I missed this one. I absorbed your fixes and rebased onto master. Hopefully this should fix it, but would be great if I could view the build logs. Tried to log in and authenticate through GitHub but didn't have authorization to make an account. Any idea here?

@rileykarson do you want to invoke the graphite-docker-images build trigger or should I try to test it locally?

**Comment by rileykarson** [comment]
Ah- those logs need to be locked down, sorry. They've got access to credentials so we need to protect against changes that expose them.

Can you rebase on top of master and push again? The parent of your commit is https://github.com/GoogleCloudPlatform/magic-modules/commit/464a79dc8b1d6564d7a7ccdc6ca90c5d36d1739b which includes some old build parameters that aren't supported by our current build images.

**Comment by rsinnet** [comment]
> Ah- those logs need to be locked down, sorry. They've got access to credentials so we need to protect against changes that expose them.
> 
> Can you rebase on top of master and push again? The parent of your commit is [464a79d](https://github.com/GoogleCloudPlatform/magic-modules/commit/464a79dc8b1d6564d7a7ccdc6ca90c5d36d1739b) which includes some old build parameters that aren't supported by our current build images.

Whoops! I had forgotten to pull in the upstream 🙈 Fixed. Thanks, @rileykarson

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb [hunk 6]
```diff
@@ -31,6 +31,14 @@ var ignoredReplicaConfigurationFields = []string{
 	"deletion_protection",
 }
 
+var ignoredSourceRepresentationFields = []string{
+	"ca_certificate",
```
Comment (rsinnet) [inline mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:35]: Assuming these should be snake case based on the above lines.

File: mmv1/products/sql/terraform.yaml [hunk 7]
```diff
@@ -62,6 +72,10 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       onPremisesConfiguration.port: !ruby/object:Overrides::Terraform::PropertyOverride
         validation: !ruby/object:Provider::Terraform::Validation
           function: 'validation.IntBetween(0, 65535)'
+      onPremisesConfiguration.password: !ruby/object:Overrides::Terraform::PropertyOverride
+        sensitive: true
+      onPremisesConfiguration.client_key: !ruby/object:Overrides::Terraform::PropertyOverride
```
Comment (rileykarson) [inline mmv1/products/sql/terraform.yaml:77]:
```suggestion
      onPremisesConfiguration.clientKey: !ruby/object:Overrides::Terraform::PropertyOverride
```

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb (hunk 6 above)
Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:35]: Yep!

**Review by rileykarson — COMMENTED** [review]
Another build failure, see inline

File: mmv1/products/sql/terraform.yaml (hunk 7 above)
Comment (rsinnet) [reply mmv1/products/sql/terraform.yaml]: Thanks!

Comment (rsinnet) [reply mmv1/products/sql/terraform.yaml]: > Another build failure, see inline

Thanks @rileykarson. Fixed!

Sorry you have to play message router 🙈

**Comment by rsinnet** [comment]
> 

@rileykarson I think I got the issues? Didn't see any other inline?

**Comment by rileykarson** [comment]
Hmm- I can't see the issue in the logs as they're too large, likely due to an issue outside of this PR. Do you mind rebasing on top of the current `master`? That should let me get the error logs.

**Comment by rsinnet** [comment]
> Hmm- I can't see the issue in the logs as they're too large, likely due to an issue outside of this PR. Do you mind rebasing on top of the current `master`? That should let me get the error logs.

@rileykarson rebased onto master!

File: mmv1/products/sql/api.yaml (hunk 1 above)
Comment (rileykarson) [reply mmv1/products/sql/api.yaml:748]: Ah- this is the cause of the lint failures as well. Are you able to apply this change?

**Pull request closed without merging**